packages/foam-vscode/src/features/commit-on-save.ts

```diff
@@ -0,0 +1,100 @@
+import { commands, ExtensionContext, TextDocument, workspace, extensions } from "vscode";
+import { GitExtension, Commit } from "../git";
+import { FoamFeature } from "../types";
+import { exec } from 'child_process';
+import { promisify } from 'util';
+
+const execP = promisify(exec);
+let latestCommitTime;
+let latestSyncTime;
+let commitInterval = 10; // replace with configuration
+let syncInterval = 60 * 5;
+
+const feature: FoamFeature = {
+  activate: (context: ExtensionContext) => {
+    context.subscriptions.push(
+      commands.registerCommand("foam-vscode.run-hook-on-save", commitOnSave)
+    );
+    workspace.onDidSaveTextDocument((document: TextDocument) => {
+      commands.executeCommand("foam-vscode.run-hook-on-save", document);
+    });
+  }
+};
+
+async function commitOnSave (document: TextDocument) {
+  let amend = false;
+
+  const gitExtension = extensions.getExtension<GitExtension>('vscode.git').exports;
```

Comment:
is this the git built-in extension in vscode? what I mean is, is it possible it would not be defined?

Reply:
Yes, it is built-in, but it possible that user disabled it manually. We should throw an error if it is undefined.